src/Controller/ApiController.php

```diff
@@ -58,7 +58,7 @@ public function process()
      */
     public function listing()
     {
-        $this->request['service'] = 'listing';
+        $this->request->withParam('service', 'listing');
```

Comment:
thats wrong as withParam  returns an updated request object. This method returns  * a *new* request object and does not mutate the request in-place. So you should reasign $this->request. 
correct code is ```$this->request = $this->request->withParam('service', 'listing');```